views/login.tpl.html

```diff
@@ -1,55 +1,183 @@
+  <script>
+    document.addEventListener('DOMContentLoaded', function () {
+      const loginForm = document.getElementById('login-form');
+      const usernameInput = document.getElementById('username');
+      const passwordInput = document.getElementById('password');
+      const loginBtn = document.getElementById('login-btn');
+
+      // Check WebAuthn support
+      const webauthnSupported = WebAuthnUtil.isSupported();
+
+      if (!webauthnSupported) {
+        // If WebAuthn is not supported, make password required and fallback to traditional login
+        passwordInput.required = true;
+      }
+
+      // Helper function to get appropriate error message
+      function getWebAuthnErrorMessage(error) {
+        const baseMessage = 'Passkey authentication failed. ';
+
+        if (error.message.includes('No passkey registered for this username')) {
+          return baseMessage + 'No passkey found for this username. Please enter your password or register a passkey first in your account settings.';
+        }
+        if (error.message.includes('User not found')) {
+          return baseMessage + 'Username not found. Please check your username and try again.';
+        }
+        if (error.message.includes('Too many requests')) {
+          return baseMessage + 'Too many login attempts. Please wait a moment and try again.';
+        }
+        if (error.message.includes('NotAllowedError')) {
+          return baseMessage + 'Operation was cancelled or timed out.';
+        }
+        if (error.message.includes('SecurityError')) {
+          return baseMessage + 'Security error occurred.';
+        }
+        if (error.message.includes('InvalidStateError')) {
+          return baseMessage + 'Security key is already registered.';
+        }
+        if (error.message.includes('NotSupportedError')) {
+          return baseMessage + 'This security key is not supported.';
+        }
+        if (error.message.includes('invalid credentials') || error.message.includes('authentication failed')) {
+          return baseMessage + 'No passkey found for this username. Please enter your password or register a passkey.';
+        }
+        return baseMessage + (error.message || 'Please try again with your password.');
+      }
+
+      // Helper function to handle WebAuthn authentication
+      async function handleWebAuthnLogin(username, loginBtn) {
+        const originalBtnText = loginBtn.innerHTML;
+        loginBtn.disabled = true;
+        loginBtn.innerHTML = '<svg class="inline w-4 h-4 mr-2 animate-spin" fill="currentColor" viewBox="0 0 24 24"><circle cx="12" cy="12" r="10" stroke="currentColor" stroke-width="4" fill="none" stroke-dasharray="31.416" stroke-dashoffset="31.416"><animate attributeName="stroke-dasharray" dur="2s" values="0 31.416;15.708 15.708;0 31.416" repeatCount="indefinite"/><animate attributeName="stroke-dashoffset" dur="2s" values="0;-15.708;-31.416" repeatCount="indefinite"/></circle></svg>Signing in...';
```

Comment:
I don't think this is very good style. For simplicity, simply keep the button as is, I don't think we need this intermediate "loading" state.